plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -1039,4 +1076,16 @@ public boolean isTableBucketEnabled(int tableBucketNumber)
             return bucketFilter.test(tableBucketNumber);
         }
     }
+
+    private static void checkExecutorIsNotDirectExecutor(Executor executor)
+    {
+        ReentrantLock lock = new ReentrantLock();
+        lock.lock();
+        try {
+            executor.execute(() -> checkState(!lock.isHeldByCurrentThread(), "executor is a direct executor"));
```

Comment:
Do we actually need the lock? This seems to be dependent on the fact that the exception is thrown in the calling thread, so we could simply do
```java
executor.execute(() -> {
    throw new IllegalArgumentException("executor is a direct executor");
});

Reply:
I thought about that. I'm worried that if an executor is configured to handle uncaught exceptions (for example log them) it will create unnecessary noise in the output.

Reply:
![image](https://user-images.githubusercontent.com/5570988/194290455-6404a9be-9782-4ee8-9a84-09b8f7989dff.png)